### PR TITLE
Fixed Issue #25126:Newletter subscription issue

### PR DESCRIPTION
**Pull request by ghost**
### Fixed Issues (if relevant)

1. magento/magento2#25126: Newletter subscription issue

### Manual testing scenarios (*)

1. Go to any page on frontend and find newsletter section at bottom right of the page.
2. Login into admin panel.
3. Disable Newsletter. Navigate to Store -> Configuration -> Customer -> Newsletter -> General Options and set enable to **NO**
4. Now go to opened page on frontend and try to subscribe.
5. Now there will be no route page.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ghost** [comment]
@melnikovi @buskamuza - please review this pull request

**Comment by ghost** [comment]
@melnikovi @buskamuza - Is there any changes required from my end for this pull request. Please let me know so that i can update it accordingly.

**Comment by melnikovi** [comment]
I think it might be good idea to hide form as well, need to confirm with PO. I will follow up with community engineering team.

File: app/code/Magento/Newsletter/Controller/Subscriber.php [hunk 1]
```diff
@@ -41,24 +49,67 @@ abstract class Subscriber extends \Magento\Framework\App\Action\Action
      */
     protected $_customerUrl;
 
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected $_scopeConfig;
```
Comment (melnikovi) [inline app/code/Magento/Newsletter/Controller/Subscriber.php]: Properties should be private if there are no reasons to make them protected. Also, by naming convention we don't use underscore in naming.

Comment (ghost) [reply app/code/Magento/Newsletter/Controller/Subscriber.php]: @melnikovi - I have updated the variable name and updated the scope of possible variables.

**Comment by ghost** [comment]
@melnikovi - Hi! regarding Integration Tests
I try to run integration testing in my local instance. Test was successful. I tried 2 times.
But here it is showing failed integration test .
Can you let me know the environment details so that i can check this.?
Or can you help me with integration test?

**Comment by ghost** [comment]
@melnikovi - I have implement the requested changes. Please review pull request and let me know there is any update required from my end.

**Review by rodrigowebjump — CHANGES_REQUESTED** [review]
Hi @rahulwebkul Thanks for your PR.

Please revert refactor of properties declaration. 

Try to keep the changes as minimum as possible. 
As you can see changing this properties forces you to refactor many class of core and also perhaps my third part modules will became incompatible.

Your PR should fix specifically the check if feature is enabled in admin.

Thanks

**Comment by ghost** [comment]
@rodrigowebjump - I updated the other files due to some miscommunication between me and @melnikovi .
Now i have updated the only file and restored the other files to its original state.
I have also added backward compatibility. So there will be no issue in child classes. 
Please check and approve the changes.

File: app/code/Magento/Newsletter/Controller/Subscriber.php [hunk 2]
```diff
@@ -3,20 +3,29 @@
...
 
+/**
+ * Newsletter subscribe controller
+ */
 abstract class Subscriber extends \Magento\Framework\App\Action\Action
 {
+    /**
+     * Configuration path to newsletter active setting
+     */
+    const XML_PATH_NEWSLETTER_ACTIVE = 'newsletter/general/active';
```
Comment (VladimirZaets) [inline app/code/Magento/Newsletter/Controller/Subscriber.php]: Please add scope to constant. In this case it should be private

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Please take a look at my review comments

File: app/code/Magento/Newsletter/Controller/Subscriber.php (hunk 2 above)
Comment (ghost) [reply app/code/Magento/Newsletter/Controller/Subscriber.php]: @VladimirZaets - I have added the scope in constant. Please check.

**Comment by VladimirZaets** [comment]
Hi @rahulwebkul . Thanks for collaboration. Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by automated tests.

**Comment by engcom-Charlie** [comment]
I'll take care of automated tests.

**Comment by rodrigowebjump** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, here is your [new Magento instance](https://pr-25127.instances.magento-community.engineering).
Admin access: [https://pr-25127.instances.magento-community.engineering/admin_d41f](https://pr-25127.instances.magento-community.engineering/admin_d41f)
Login: `8af943a5` Password: `5361fb5d2244`
Instance will be terminated in up to 3 hours.

File: app/code/Magento/Newsletter/Controller/Subscriber.php [hunk 3]
```diff
@@ -3,20 +3,29 @@
...
 
+/**
+ * Newsletter subscribe controller
+ */
 abstract class Subscriber extends \Magento\Framework\App\Action\Action
 {
+    /**
+     * Configuration path to newsletter active setting
+     */
+    private const XML_PATH_NEWSLETTER_ACTIVE = 'newsletter/general/active';
```
Comment (lenaorobei) [inline app/code/Magento/Newsletter/Controller/Subscriber.php:27]: Please see `\Magento\Newsletter\Model\Config::isActive`.

**Review by lenaorobei — COMMENTED** [review]
Please update the feature branch and see review comment.

**Comment by lenaorobei** [comment]
Closing this PR due to inactivity. Please feel free to reopen if you are going to continue working on it.
Thank you.

**Pull request closed without merging**